### ISSUE TITLE
Why is currentItem `nil` when playlist played to the end?

### ISSUE THREAD

**Issue by mbruegmann** [SRGSSR/pillarbox-apple#1030]
### Can your answer be found elsewhere?

- [x] I have searched existing [issues](https://github.com/SRGSSR/pillarbox-apple/issues?q=is%3Aissue+label%3Aquestion+) and found no similar question.

- [x] I have browsed the available documentation and found no answer to my question.

### Detailed question

When we play a playlist to its end it seems the players currentItem is `nil`. 
But anyway `canReplay()` returns true and we can restart that item. 
Shouldn't currentItem still be set to the last item or maybe to the first item? This way we could easily display to the user which item was just played or which will be played once we call `replay()`?

**Comment by defagos** [comment]
As discussed with you this is a tricky behavior. Some apps might want to display nothing, some apps might want to stick with the last content being played or, as in your case, some apps might want to display the first item in this case.

The behavior is tricky and introducing a behavioral parameter to govern the player behavior for something that is more UI-related in the end would probably lead to a lot of complexity. On the app side, though, the code required is mostly:

```swift
var displayedItem: PlayerItem? {
    currentItem ?? items.first
}
````

or

```swift
var displayedItem: PlayerItem? {
    currentItem ?? items.last
}
```

since item / current item changes are automatically observed by `Player` and lead to view bodies being recalculated.

TL;DR Better do nothing and let the app decide. Flexible and much simpler / safer for everyone.
